common/src/main/java/org/apache/uniffle/common/provider/HadoopAccessorProvider.java

```diff
@@ -0,0 +1,265 @@
+      LOGGER.error("Fail to get filesystem of {}", path);
+      throw e;
+    }
+  }
+
+  /**
+   * The only entrypoint is to get the hadoop filesystem instance and is compatible with
+   * the kerberos security.
+   *
+   * When to invoke this method?
+   * 1. For shuffle server side, it needs to get filesystem before writing the shuffle data to secured HDFS
+   * with the spark job's user auth.
+   *
+   * @param user
+   * @param path
+   * @param conf
+   * @return
+   * @throws Exception
+   */
+  public static FileSystem getFilesystem(
+      final String user,
+      final Path path,
+      final Configuration conf) throws Exception {
+    UserGroupInformation.AuthenticationMethod authenticationMethod =
+        SecurityUtil.getAuthenticationMethod(conf);
+    boolean securityEnable = authenticationMethod != UserGroupInformation.AuthenticationMethod.SIMPLE;
+    return getInternalFileSystem(user, securityEnable, path, conf);
+  }
+
+  /**
+   * The method is to return the Hadoop Filesystem directly which is not retrieved by
+   * ugi proxy user.
+   *
+   * When to invoke this method?
+   * 1. In client side, spark shuffle-reader getting filesystem before reading shuffle data stored in HDFS.
+   * 2. In shuffle-server/coordinator side, it reads the config file stored in HDFS.
+   *
+   * @param path
+   * @param configuration
+   * @return
+   * @throws Exception
+   */
+  public static FileSystem getFileSystem(
+      final Path path,
+      final Configuration configuration) throws Exception {
+    return getInternalFileSystem(null, false, path, configuration);
+  }
+
+  /**
+   * For kerberized cluster access
+   * @param securityInfo
+   * @throws Exception
+   */
+  public static void init(SecurityInfo securityInfo) throws Exception {
+    if (provider == null) {
+      synchronized (HadoopAccessorProvider.class) {
+        if (provider == null) {
+          final HadoopAccessorProvider hadoopAccessorProvider = new HadoopAccessorProvider(securityInfo);
+          provider = hadoopAccessorProvider;
+        }
+      }
+    }
+    LOGGER.info("The {} has been initialized, kerberos enable: {}",
+        HadoopAccessorProvider.class.getSimpleName(),
+        provider.kerberosEnabled);
+  }
+
+  /**
+   * No need to
+   * For non-kerberized cluster access, like client side reading shuffle data.
+   */
+  public static void init() throws Exception {
+    init(null);
+  }
+
+  @VisibleForTesting
+  public static void cleanup() throws Exception {
+    if (provider != null) {
+      provider.close();
+      provider = null;
+    }
+  }
+
+  @Override
+  public void close() throws IOException {
+    if (cache != null) {
+      for (UserGroupInformation ugi : cache.values()) {
+        try {
+          FileSystem.closeAllForUGI(ugi);
+        } catch (IOException ioException) {
+          LOGGER.error("Exception occurred while closing filesystems for {}", ugi.getUserName(), ioException);
+        }
+      }
+      cache.clear();
```

Comment:
If there are many users, cache may occupy too much memory.

Reply:
No cache is also OK. But i think the number of users may be not too much.

Reply:
A shuffle server is a long time service, I worried about the problems, more and more people use the service, user occupy too much memory.

Reply:
If we want to use a cache, I think we need a LRU cache.

Reply:
Got it. I think no cache is also OK. I will remove it.

Reply:
Removed.